app/ydoc-shared/src/ast/tree.ts

```diff
@@ -1893,6 +1893,16 @@ export class MutableTextLiteral extends TextLiteral implements MutableExpression
     this.fields.set('close', unspaced(Token.new(code)))
   }
 
+  fixBoundaries() {
+    const open = this.open
+    const close = this.close
+    if (open != null && close == null) {
```

Comment:
If the literal is multiline, `close` *must* remain `null`. It is multiline if `open.length > 1`.

This function could use some unit tests.

Reply:
So I guess `setBoundaries` also needs this logic?

Reply:
`setBoundaries` is a very low-level operation that may break the validity of the AST in various ways, we might remove it or comment it as only for use in tests.